### PR TITLE
Connect via unix socket instead of filesystem to be compatible with php open_basedir restrictions.

### PR DESCRIPTION
**Pull request by jazzsequence**
copied from #425 to run CI workflows by @hj-collab

Behat fixes ported to https://github.com/pantheon-systems/wp-redis/pull/429/

### CODE REVIEW

**Review by pwtyler — DISMISSED** [review]

**Review by pwtyler — CHANGES_REQUESTED** [review]
Need to update README (c/f #423)

File: tests/behat/features/load-wp.feature
Comment (jazzsequence) [inline tests/behat/features/load-wp.feature:25]: spacing
```suggestion
    When I wait "1" second
```

File: tests/behat/features/wp-redis.feature
Comment (jazzsequence) [inline tests/behat/features/wp-redis.feature:23]: more spacing issues

**Comment by pwtyler** [comment]
This PR ended up all over the place and ultimately derailed on the testing. Going to close and re-open the original PR (although we still need to prioritize the work on the teamside). As we've learned since this was attempted, pushing fork PRs to a separate branch (rather than opening a new PR) stills gives us the circle checks in the original PR. 

Better/cleaner would be to move all tests to Github actions triggered "On PR", which can be approved by a codeowner to run CI after manually reviewing.

**Pull request closed without merging**
